andela/octopus-bn-backend
# #170947433 Implement user registration

#### What does this PR do?

- Implement user registration feature
- Use a 3rd API to send account verification emails.

#### How should this be manually tested?

- clone this repo
- install the dependencies by running `npm i` in your terminal.
- by referencing .env.example create a .env file and add all the keys.
- run `npm run migrateDb` in your terminal to run the migrations
- run `npm run dev` in your terminal to start the server.
- use postman to test this route `/api/v1/auth/signup` you will have to provide the following fields: firstName, lastName, email, password.
- check your email inbox, you will find an email from barefoot nomad.

#### Implementation plan

[User registration](https://docs.google.com/document/d/1qByH2hXJoWXIr9jFQIi3Co-gRLSjU_amzJQ3PnHsHDk/edit)

#### Stories associated with PR

[#170947433](https://www.pivotaltracker.com/story/show/170947433)

#### Screenshots if any
<img width="862" alt="Screen Shot 2020-02-11 at 13 31 22" src="https://user-images.githubusercontent.com/53453380/74233246-0130b080-4cd3-11ea-9843-0d7ca82928e5.png">
<img width="1108" alt="Screen Shot 2020-02-11 at 13 32 28" src="https://user-images.githubusercontent.com/53453380/74233260-042ba100-4cd3-11ea-861d-ddd4ad3d29eb.png">
